File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/ProfilePicture.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "profile_picture")
public class ProfilePicture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "picture_id")
    private Long pictureId;

    @Column(name = "picture", nullable = false)
    private String picture;

    @OneToOne
    @JoinColumn(name = "user_id", unique = true)
    private User user;
}

File: Frontend/manga-world/src/Model/ProfilePicture.ts
import User from "./User";

class ProfilePicture {
  pictureId: number;
  picture: string;
  user: User;

  constructor(pictureId: number, picture: string, user: User) {
    this.pictureId = pictureId;
    this.picture = picture;
    this.user = user;
  }
}

export default ProfilePicture;
